### PR TITLE
TeamStore migration Part 3

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

This is the las part of the migration of `TeamStore` to return plain errors.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15126

#### Jira Ticket

https://mattermost.atlassian.net/browse/MM-27319

### CODE REVIEW

File: app/syncables.go [hunk 1]
```diff
@@ -204,23 +204,20 @@ func (a *App) SyncSyncableRoles(syncableID string, syncableType model.GroupSynca
...
 
-	var updateFunc func(string, []string) *model.AppError
-
 	switch syncableType {
 	case model.GroupSyncableTypeTeam:
-		updateFunc = a.Srv().Store.Team().UpdateMembersRole
+		nErr := a.Srv().Store.Team().UpdateMembersRole(syncableID, permittedAdmins)
+		if nErr != nil {
+			// Review: should we change the key "store.update_error" to "app.update_error"? It is very general and changing it now will modify lots of files
+			return model.NewAppError("SyncSyncableRoles", "store.update_error", nil, nErr.Error(), http.StatusInternalServerError)
```
Comment (agnivade) [inline app/syncables.go:212]: We can keep a TODO and change this in a separate PR.

**Review by agnivade — APPROVED** [review]
Looks great!

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @rvillablanca! :sparkles:

**Pull request merged**